design-proposals/persist-vmi-firmware-uuid.md

```diff
@@ -0,0 +1,108 @@
+# Overview
+This proposal introduces a mechanism to persist the firmware UUID of a Virtual Machine in KubeVirt.
+By storing the firmware UUID, we ensure that it remains consistent across VM restarts.
+which is crucial for applications and services that rely on the UUID for identification or licensing purposes.
+
+## Motivation
+* Duplicate UUIDs cause burden with third-party automation frameworks such as Gitops tools, making it difficult to maintain unique VM UUID. 
+  Such tools have to manually assign unique UUIDs to VMs, thus also manage them externally. (issue)
+* The non-persistent nature of UUIDs may trigger redundant cloud-init setup upon VM restore. (issue)
+* The non-persistent nature of UUIDs may trigger redundant license activation events during VM lifetime i.e. Windows key activation re-occur when power off and power on the VM.
+
+## Goals
+* Improve the uniqueness of the VMI firmware UUID to become independent of the VM name and namespace.
+* Maintain the UUID persistence over the VM lifecycle.
+* Maintain backward compatibility
+* Maintain UUID persistence with VM backup/restore.
+
+## Non Goals
+
+
+## Definition of Users
+VM owners: who require consistent firmware UUIDs for their applications.
+
+## User Stories
+### Supported cases:
+* Creating and Starting a New VM
+* starting and old VM that was not running
```

Comment:
do we want to support that?
how would the controller distinguish between VMs that have started and VMs that have not?

Reply:
we would use the legacy uuid calculation for both running/stopped vms.